hack/release.sh

```diff
@@ -0,0 +1,107 @@
+#!/usr/bin/env bash
+# Copyright 2018 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -euo pipefail
+
+# The script will try to checkout the release version if specified.
+# This is useful for manual or older tagged version releases.
+# If a release version is not specified, it will check the latest commit
+# for a tag, if a tag has appeared on the latest commit, it will build
+# and push the binaries from the latest commit.
+RELEASE_VERSION=${1:-}
+RELEASE_TAG=${RELEASE_VERSION:-}
+CHECKOUT_OLDER="${2:-}"
+GCP_PROJECT=${GCP_PROJECT:-k8s-jkns-e2e-gce-federation}
+GCS_BUCKET=${GCS_BUCKET:-kubernetes-federation-release}
+GCR_REPO_PATH="gcr.io/google_containers/fcp-amd64"
+SCRIPT_ROOT="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+TMPDIR="$(mktemp -d /tmp/k8s-fed-relXXXXXXXX)"
+KUBE_ROOT=${SCRIPT_ROOT}/..
+
+source "${KUBE_ROOT}/hack/lib/version.sh"
+
+function clean_up() {
+  if [[ "${TMPDIR}" == "/tmp/k8s-fed-rel"* ]]; then
+    rm -rf "${TMPDIR}"
+  fi
+}
+trap clean_up EXIT
+
+cd "${SCRIPT_ROOT}/.."
```

Comment:
nit: can use `KUBE_ROOT` here. Also better to use either `cd` or `pushd` consistently within a script.

Reply:
Done!